apiserver/facades/client/machinemanager/upgradeseries.go

```diff
@@ -464,7 +465,11 @@ func (s stateSeriesValidator) verifySupportedBase(application Application, base
 		return errors.Trace(err)
 	}
 	if len(supportedSeries) == 0 {
-		supportedSeries = append(supportedSeries, ch.URL().Series)
+		curl, err := charm.ParseURL(ch.URL())
+		if err != nil {
+			return errors.Trace(err)
+		}
+		supportedSeries = append(supportedSeries, curl.Series)
 	}
```

Comment:
same regarding charms with no series. - That should be in a separate commit? Or explained somewhere

Reply:
I'll remove this in a separate commit